src/lib/mongoBackend/mongoUpdateContextSubscription.cpp

```diff
@@ -56,487 +46,43 @@ HttpStatusCode mongoUpdateContextSubscription
-                               CSUB_ENTITY_TYPE << en->type <<
-                               CSUB_ENTITY_ISPATTERN << en->isPattern));
-        }
-      }
-      newSub.append(CSUB_ENTITIES, entities.arr());
-    }
-    else
-    {
-      newSub.appendArray(CSUB_ENTITIES, getFieldF(sub, CSUB_ENTITIES).Obj());
-    }
-
-    // Attributes
-    if (requestP->attributeList.size() > 0)
-    {
-      /* Build attributes array */
-      BSONArrayBuilder attrs;
-      for (unsigned int ix = 0; ix < requestP->attributeList.size(); ++ix) {
-        attrs.append(requestP->attributeList[ix]);
-      }
-      newSub.append(CSUB_ATTRS, attrs.arr());
-    }
-    else
-    {
-      newSub.appendArray(CSUB_ATTRS, getFieldF(sub, CSUB_ATTRS).Obj());
-    }
-  }
+  requestP->toNgsiv2Subscription(&sub);
+  std::string subId = mongoUpdateSubscription(sub, &oe, tenant, servicePathV, xauthToken, fiwareCorrelator);
 
-  /* Expiration */
-  long long expiration = sub.hasField(CSUB_EXPIRATION)? getIntOrLongFieldAsLongF(sub, CSUB_EXPIRATION) : -1;
-  if (version == "v1")
+  if (subId != "")
   {
-    // Based on duration
+    // Duration and throttling are optional parameters, they are only added in the case they were used for update
     if (!requestP->duration.isEmpty())
     {
-      expiration = getCurrentTime() + requestP->duration.parse();
+      responseP->subscribeResponse.duration = requestP->duration;
     }
-  }
-  else // v2
-  {
-    // Based on expires
-    if (requestP->expires > 0)
-    {
-      expiration = requestP->expires;
-    }
-  }
-  newSub.append(CSUB_EXPIRATION, expiration);
-  LM_T(LmtMongo, ("New subscription expiration: %ld", (long) expiration));
 
-  /* ServicePath update */
-  newSub.append(CSUB_SERVICE_PATH, servicePathV[0] == "" ? DEFAULT_SERVICE_PATH_QUERIES : servicePathV[0]);
-
-  /* Throttling update */
-  if (!requestP->throttling.isEmpty())
-  {
-    /* Throttling equal to 0 removes throttling */
-    long long throttling = requestP->throttling.parse();
-
-    if (throttling != 0)
+    if (!requestP->throttling.isEmpty())
     {
-      newSub.append(CSUB_THROTTLING, throttling);
+      responseP->subscribeResponse.throttling = requestP->throttling;
     }
-  }
-  else if (requestP->throttling.seconds > 0) // v2
-  {
-      newSub.append(CSUB_THROTTLING, (long long) requestP->throttling.seconds);
+    responseP->subscribeResponse.subscriptionId = subId;
   }
   else
   {
-    //
-    // The hasField check is needed as Throttling might not be present in the original doc
-    // FIXME P1: However, we could include Throttling in the new doc anyway ...
-    //
-    if (sub.hasField(CSUB_THROTTLING))
+    // Check OrionError. Depending the error kind, details are included or not in order
+    // to have a better backward compatiblity
+    // FIXME: should we? or it is better modify .test and provide more accurate errors with 'details'?
+    if (oe.code == SccContextElementNotFound)
```

Comment:
Maybe is better not doing exceptions and modifying the .test adding the extra "detail" field. In some way, it would be a kind of "fixing", as now we can provide more info about the error.

What do you think?